pkg/binary/envoy/controlplane/istio.tmpl_test.go

```diff
@@ -19,48 +19,46 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
-	"os"
-	"path/filepath"
 	"testing"
 
-	"github.com/mholt/archiver"
+	"github.com/mholt/archiver/v3"
 	"github.com/stretchr/testify/require"
 )
 
 // TestIstioBootStrapTemplateEqualsReleaseJson ensures istioBootStrapTemplate is exactly the same as what would have been
 // downloaded from the istio release for version IstioVersion.
 func TestIstioBootStrapTemplateEqualsReleaseJson(t *testing.T) {
-	// Retrieve Istio tarball os doesn't matter we only car about the bootstrap JSON
+	// Retrieve Istio tarball os doesn't matter we only care about the bootstrap JSON
 	url := fmt.Sprintf("https://github.com/istio/istio/releases/download/%s/istio-%s-linux.tar.gz", IstioVersion, IstioVersion)
 	resp, err := http.Get(url)
 	require.NoError(t, err, "error getting tarball for istio version %s", IstioVersion)
 
 	defer resp.Body.Close() //nolint
 	require.Equal(t, http.StatusOK, resp.StatusCode, "unexpected HTTP status from %s", url)
 
-	dst, err := ioutil.TempDir("", "")
-	require.NoError(t, err, `ioutil.TempDir("", "") erred`)
-	defer os.RemoveAll(dst)
+	tarGz := archiver.NewTarGz()
+	defer tarGz.Close() //nolint
 
-	tarball := filepath.Join(dst, "istio.tar.gz")
-	f, err := os.OpenFile(tarball, os.O_CREATE|os.O_WRONLY, 0600)
-	require.NoError(t, err, "Couldn't open file %s", tarball)
-	defer f.Close() //nolint
+	err = tarGz.Open(resp.Body, int64(0))
+	require.NoError(t, err, "error reading TarGz from %s", url)
 
-	_, err = io.Copy(f, resp.Body)
-	require.NoError(t, err, "Couldn't download %s into %s", url, tarball)
-
-	// Walk the tarball until we find the bootstrap
 	bootstrapName := "envoy_bootstrap_v2.json"
 	var bytes []byte
-	err = archiver.Walk(tarball, func(f archiver.File) error {
+	for {
```

Comment:
I switched to streaming style as even though there aren't that many less lines here, I feel it is easier to reason with vs moving contents into various directories.

It is disappointing we need this test, as it makes our unit tests perform I/O against github...

Reply:
@mathetake I was thinking.. what if we rewrote this as a couple lines of shell script and checked as a part of `make check` instead of a "unit test". Another option is to remove it entirely, saving the 30MB download every time tests run.